pipeline/01-train.R

```diff
@@ -170,6 +169,23 @@ lgbm_model <- parsnip::boost_tree(
     lambda_l2 = tune()
   )
 
+# The num_iterations (trees) parameter has three possible values:
+# 1. If CV AND early stopping are enabled, set a static parameter using the
+#    upper bound of the CV search range as the maximum possible number of trees
+# 2. If CV is enabled but early stopping is disabled, set the search range to
+#    the standard CV range
+# 3. If no CV is enabled, use the default parameter value
+if (cv_enable && early_stopping_enable) {
```

Comment:
**[Question, non-blocking]** Is there a reason to do this check here instead of folding it into the existing `if (cv_enable)` conditional block starting on line 204?

Reply:
Yes! The model specification gets rolled into a workflow on L192, and the setup of the workflow needs to happen regardless of whether CV is going to run.